action/staking/stake.go

```diff
@@ -134,6 +136,41 @@ func (s stakeTx) ProcessFee(ctx *action.Context, signedTx action.SignedTx, start
 	return action.BasicFeeHandling(ctx, signedTx, start, size, 2)
 }
 
+// allow validator's stake address update only when it's clean
+func isStakeAddressClean(ctx *action.Context, v *identity.Validator) (bool, error) {
+	options, err := ctx.GovernanceStore.GetStakingOptions()
+	if err != nil {
+		return false, err
+	}
+	zero := balance.NewAmountFromInt(0)
+
+	// check locked amount
+	lockedAmt, err := ctx.Delegators.GetValidatorDelegationAmount(v.Address, v.StakeAddress)
+	if err != nil {
+		return false, err
+	}
+	if !lockedAmt.Equals(*zero) {
+		return false, nil
+	}
+
+	// check pending amount
+	pendingAmounts := ctx.Delegators.GetMaturedPendingAmount(v.StakeAddress, ctx.Header.Height, options.MaturityTime+1)
+	if len(pendingAmounts) != 0 {
+		return false, nil
+	}
+
+	// check bounded amount
+	boundCoin, err := ctx.Delegators.GetDelegatorBoundedAmount(v.StakeAddress)
```

Comment:
What does the bounded amount mean?

Reply:
@KevinOneLedger I was initially confused by the word 'bounded'. I figured out from his implementation that the word 'bounded' here is the same as 'withdrable' amount.